tests/gradients/test_finite_difference.py

```diff
@@ -373,9 +373,12 @@ def __mul__(self, other):
                 return SpecialObject(self.val * other)
 
             def __add__(self, other):
-                new = self.val + other.val if isinstance(other, self.__class__) else other
+                new = self.val + (other.val if isinstance(other, self.__class__) else other)

```

Comment:
This was a silent bug: The addition previously was executed _before_ the Boolean clause, yielding `new=other` for the case of `not isinstance(other, self.__class__)`.